data/data.go

```diff
@@ -120,18 +120,22 @@ func nextThree(events []Event) []Event {
 	return events[0:3]
 }
 
-func events(name string) ([]Event, error) {
+func events(name string) (data []Event, err error) {
 	resp, err := http.Get(fmt.Sprintf(apiTemplate, name))
+
 	if err != nil {
-		log.Fatal(err)
+		return data, err
 	}
+
 	defer resp.Body.Close()
 
 	decoder := json.NewDecoder(resp.Body)
-	var data []Event
+
 	err = decoder.Decode(&data)
+
 	if err != nil {
```

Comment:
This entire thing can _could_ be collapsed down to:

```go
return data, decoder.Decode(&data)
```

However I prefer the following for clarity:

```go
err = decoder.Decode(&data)
if err != nil {
    return nil, err
}
return data, nil
```

It makes the two code paths clear and is less to think about when reading it. I know if an error occurs I will get `nil` back for `data`. That's mostly style however.

Reply:
There's a subtle difference here between what I did and what you're suggesting though, right?  In the snippet above, you're explicitly returning `nil`, where I am either returning a zero'd out slice or potentially garbage data if `json.Decode` fails.  I think explicitly returning `nil` is better for both clarity for reading and consistency when calling.  I can submit a small PR to fix.

Reply:
I didn't mind what you did, but yeah, being explicit about the `nil`s means less to think about when debugging.